### ISSUE TITLE
Dot in field name error

### ISSUE THREAD

**Issue by komantsev** [markmcdowell/NLog.Targets.ElasticSearch#46]
Hi. I use **NLog.Targets.ElasticSearch** and I found a bug:
I log an **System.AggregateException**, who contains field **InnerExceptions** (enumerable type). **InnerExceptions** contains an **System.Data.SqlClient.SqlException** with filled **Data** field, who has Keys with dot in theirs names (ex. "**HelpLink.BaseHelpUrl**").
Extension method **ExpandoObjectExtensions.ReplaceDotInKeys** ignore enumerable types and ElasticClient send JSON with dots in field names.
I propose the following fix:
```csharp
public static ExpandoObject ReplaceDotInKeys(this ExpandoObject obj)
{
    var clone = new ExpandoObject();
    foreach (var item in obj)
    {
        if (item.Value == null)
            continue;                

        if (item.Value.GetType() == typeof(ExpandoObject))
            ((IDictionary<string, object>)clone)[item.Key.Replace('.', '_')]
                = (item.Value as ExpandoObject).ReplaceDotInKeys();
        // New code block >>>
        else if (item.Value is IEnumerable<object>)
        {
            var enumerableClone = new List<ExpandoObject>();
            foreach (var el in (item.Value as IEnumerable<object>))
            {
                if (el is ExpandoObject)
                    enumerableClone.Add(ReplaceDotInKeys(el as ExpandoObject));
            }
            ((IDictionary<string, object>)clone)[item.Key.Replace('.', '_')] = enumerableClone;
        }
        // <<< New code block
        else if (item.Key.Contains('.'))
            ((IDictionary<string, object>)clone)[item.Key.Replace('.', '_')] = item.Value;
        else
            ((IDictionary<string, object>)clone)[item.Key] = item.Value;
    }
    return clone;
}
```

**Comment by snakefoot** [comment]
See also:

https://www.elastic.co/guide/en/elasticsearch/reference/2.4/dots-in-names.html

**Comment by markmcdowell** [comment]
Should be fixed in [5.1.0](https://github.com/markmcdowell/NLog.Targets.ElasticSearch/releases/tag/5.1.0)
